blog/2022-1-2-your-first-extension.md

```diff
@@ -0,0 +1,40 @@
+---
+slug: your-first-extension
+title: Your first extension
+authors: [CharlieS1103]
+tags: [spicetify, documentation, community,development]
+---
+
+## So you want to make an extension....
+
+
+### First step:
+Start out by setting up your environment: 
+In terminal or powershell, run ``spicetify enable-devtool apply``
+Ensure that you have a text editor or IDE at the ready.
+
+
+### Second step:
+First, run spicetify config-dir, this should open up file explorer or finder(dependent on your OS), next, open the "Extensions folder" and create a file titled extension.js, or whichever title you choose(For this tutorial we will be using extension.js). Open this file, and input the following code:
+```js
+// The async modifier allows for the user of await, which converts a promise into an object, when not using await, async is not necessary.
+(async function extension() {
+    //The following code segment waits for platform to load before running the code, this is important to avoid errors. When using things such as Player or URI, it is necessary to add those as well.
+    const { Platform } = Spicetify;
+    if (!(Platform)) {
+        setTimeout(extension, 300)
+        return
```

Comment:
Missing semicolons in a few spots too

Reply:
Semicolons are optional, so I had let it slide :eyes:

Reply:
But you did put the semicolon in line 21 @CharlieS1103, so please be consistent I suppose.